packages/components/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,274 @@
+import {
+  Pressable,
+  StyleProp,
+  Text,
+  View,
+  ViewStyle,
+  useWindowDimensions,
+} from 'react-native'
+import { spacing } from '@department-of-veterans-affairs/mobile-tokens'
+import { useTranslation } from 'react-i18next'
+import React, { FC } from 'react'
+
+import { ComponentWrapper } from '../../wrapper'
+import { Icon, IconProps } from '../Icon/Icon'
+import { Spacer } from '../Spacer/Spacer'
+import { useTheme } from '../../utils'
+
+export type CheckboxProps = {
+  /** Primary text for checkbox */
+  label: string
+  /** OnPress logic to alter `checked` state or other behavior associated with the checkbox */
+  onPress: () => void
+  /** True to make checkbox appear as checked */
+  checked?: boolean
+  /** Description that appears below label */
+  description?: string
+  /** Accessibility override for description text  */
+  descriptionA11y?: string
+  /** Hint text. Appears below header */
+  hint?: string
+  /** Accessibility override for hint text  */
+  hintA11y?: string
+  /** Optional error text. If present, applies error styling to checkbox */
+  error?: string
+  /** Accessibility override for error text  */
+  errorA11y?: string
+  /** Header text */
+  header?: string
+  /** Accessibility override for header  */
+  headerA11y?: string
+  /** True to apply indeterminate icon to checkbox */
+  indeterminate?: boolean
+  /** Accessibility override for label  */
+  labelA11y?: string
+  /** True to append (*Required) suffix to label */
+  required?: boolean
+  /** True to apply tile styling */
+  tile?: boolean
+}
+
+export const Checkbox: FC<CheckboxProps> = ({
+  checked,
+  label,
+  labelA11y,
+  description,
+  descriptionA11y,
+  error,
+  errorA11y,
+  header,
+  headerA11y,
+  hint,
+  hintA11y,
+  indeterminate,
+  onPress,
+  required,
+  tile,
+}) => {
+  const { t } = useTranslation()
+  const theme = useTheme()
+  const fontScale = useWindowDimensions().fontScale
+
+  /**
+   * Container styling
+   */
+  let containerStyle: ViewStyle = {
+    width: '100%',
+  }
+
+  if (error) {
+    containerStyle = {
+      ...containerStyle,
+      borderLeftWidth: spacing.vadsSpace2xs,
+      borderColor: theme.vadsColorFormsBorderError,
+      paddingLeft: spacing.vadsSpaceLg,
+    }
+  }
+
+  /**
+   * Pressable styling
+   */
+  const pressableBaseStyle: StyleProp<ViewStyle> = {
+    width: '100%',
+    flexDirection: 'row',
+    alignItems: 'flex-start',
+  }
+
+  const tileStyle: ViewStyle = {
+    ...pressableBaseStyle,
+    borderWidth: 2,
+    borderRadius: spacing.vadsSpace2xs,
+    paddingTop: spacing.vadsSpaceSm,
+    paddingBottom: spacing.vadsSpaceSm,
+    paddingLeft: spacing.vadsSpaceSm,
```

Comment:
Could be
```suggestion
    padding: spacing.vadsSpaceSm,
```
as specific sides should override the general rule.